routers/api/v1/projects/project.go

```diff
@@ -0,0 +1,420 @@
+
+	if err := project_model.DeleteProjectByID(ctx, ctx.ParamsInt64(":id")); err != nil {
+		ctx.Error(http.StatusInternalServerError, "DeleteProjectByID", err)
+		return
+	}
+
+	ctx.Status(http.StatusNoContent)
+}
+
+func ListUserProjects(ctx *context.APIContext) {
+	// swagger:operation GET /user/projects project projectListUserProjects
+	// ---
+	// summary: List repository projects
+	// produces:
+	// - application/json
+	// parameters:
+	//   - name: closed
+	//     in: query
+	//     description: include closed issues or not
+	//     type: boolean
+	//   - name: page
+	//     in: query
+	//     description: page number of results to return (1-based)
+	//     type: integer
+	//   - name: limit
+	//     in: query
+	//     description: page size of results
+	//     type: integer
+	// responses:
+	//  "200":
+	//    "$ref": "#/responses/ProjectList"
+	//  "403":
+	//    "$ref": "#/responses/forbidden"
+	//  "404":
+	//    "$ref": "#/responses/notFound"
+	projects, count, err := db.FindAndCount[project_model.Project](ctx, project_model.SearchOptions{
+		Type:        project_model.TypeIndividual,
+		IsClosed:    ctx.FormOptionalBool("closed"),
+		OwnerID:     ctx.Doer.ID,
+		ListOptions: db.ListOptions{Page: ctx.FormInt("page")},
+	})
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "Projects", err)
+		return
+	}
+
+	ctx.SetLinkHeader(int(count), setting.UI.IssuePagingNum)
+	ctx.SetTotalCountHeader(count)
+
+	apiProjects, err := convert.ToAPIProjectList(ctx, projects)
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "Projects", err)
+		return
+	}
+
+	ctx.JSON(http.StatusOK, apiProjects)
+}
+
+func ListOrgProjects(ctx *context.APIContext) {
+	// swagger:operation GET /orgs/{org}/projects project projectListOrgProjects
+	// ---
+	// summary: List repository projects
+	// produces:
+	// - application/json
+	// parameters:
+	//   - name: org
+	//     in: path
+	//     description: owner of the repository
+	//     type: string
+	//     required: true
+	//   - name: closed
+	//     in: query
+	//     description: include closed issues or not
+	//     type: boolean
+	//   - name: page
+	//     in: query
+	//     description: page number of results to return (1-based)
+	//     type: integer
+	//   - name: limit
+	//     in: query
+	//     description: page size of results
+	//     type: integer
+	// responses:
+	//  "200":
+	//    "$ref": "#/responses/ProjectList"
+	//  "403":
+	//    "$ref": "#/responses/forbidden"
+	//  "404":
+	//    "$ref": "#/responses/notFound"
+	projects, count, err := db.FindAndCount[project_model.Project](ctx, project_model.SearchOptions{
+		OwnerID:     ctx.Org.Organization.AsUser().ID,
+		ListOptions: db.ListOptions{Page: ctx.FormInt("page")},
+		IsClosed:    ctx.FormOptionalBool("closed"),
+		Type:        project_model.TypeOrganization,
+	})
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "Projects", err)
+		return
+	}
+
+	ctx.SetLinkHeader(int(count), setting.UI.IssuePagingNum)
```

Comment:
This seems wrong - why do we use UI page size inside API route, if it is provided by the parameter?